handbook/engineering/https_dev_instance.md

```diff
@@ -0,0 +1,173 @@
+Account                       xxxxxx (Plan: xxxxx)                                                                                                                                                                 
+Version                       2.3.35                                                                                                                                                                                    
+Region                        United States (us)                                                                                                                                                                        
+Web Interface                 http://127.0.0.1:4040                                                                                                                                                                     
+Forwarding                    http://xxxxxx.ngrok.io -> http://localhost:3080                                                                                                                                         
+Forwarding                    https://xxxxxxx.ngrok.io -> http://localhost:3080                                                                                                                                        
+                                                                                                                                                                                                                        
+Connections                   ttl     opn     rt1     rt5     p50     p90                                                                                                                                               
+                              0       0       0.00    0.00    0.00    0.00                                                                                                                                              
+```
+
+You can use the exposed HTTPS endpoint to access your instance.
+
+> SECURITY NOTE: The ngrok URLS are by definition accesible from the outside internet. Even though the URLs
+> are hard to guess anyone that guesses the URL has access to your Sourcegraph instance. It is recommended to
+> kill the ngrok proxy when you are done and not let it run for long periods of time.
+
+## Method Two
+
+> A little more involved but works without outside internet.
+
+> NOTE: On a Mac installation this method requires one restart of the machine to bounce the DNS resolving machinery
+> of the operating system. The `/etc/resolver` entry will otherwise be ignored until the next restart. 
+
+This method uses:
+
+* [dnsmasq](http://www.thekelleys.org.uk/dnsmasq/doc.html)
+* [mkcert](https://github.com/FiloSottile/mkcert)
+* [caddy2](https://caddyserver.com/)
+
+### Installing and setting up dnsmasq
```

Comment:
FYI you don't need dnsmasq to map a single address to an ip. Just edit `/etc/hosts`.

```shellsession
$ grep dev /etc/hosts
127.0.0.1 sourcegraph.example.dev
127.0.0.1 phabricator.example.dev
127.0.0.1 gitlab.example.dev
```

Reply:
yeah true :-). i guess with dnsmasq you do it once and then any .test you use resolve to 127.0.0.1. i'll add the /etc/hosts as a shorter alternative, thanks